Add stack option for CLI start command

gtsiolis opened the pull request:
<!-- Please refer to the contribution guidelines before raising a PR: https://github.com/localstack/localstack/blob/master/docs/CONTRIBUTING.md -->

<!-- Why am I raising this PR? Add context such as related issues, PRs, or documentation. -->
## Motivation

Following up from a [relevant discussion](https://localstack-cloud.slack.com/archives/C0550DW38CT/p1748416788025209), this is an attempt to add a CLI option for selecting a stack (image, emulator).

1. The command to start LocalStack for Snowflake is quite verbose by using the IMAGE_NAME environment variable. 
2. Selecting a different image version also required the IMAGE_NAME variable, see [relevant docs](https://docs.localstack.cloud/references/configuration/#core).

```sh
IMAGE_NAME=localstack/snowflake:latest localstack start
IMAGE_NAME=localstack/localstack-pro:4.1 localstack start
```

See also [relevant discussion](https://localstack-cloud.slack.com/archives/C06CV9US815/p1744792391948849) for more context.

<!-- What changes does this PR make? How does LocalStack behave differently now? -->
## Changes

This will add **--stack** and **-s** as a shorthand for selecting the image and optional a version.

```sh
localstack start                      # default, starts localstack for aws (localstack image)
localstack start --stack snowflake    # starts localstack for snowflake (snowflake image)
localstack start -s snowflake         # starts localstack for snowflake (snowflake image)
localstack start -s snowflake:0.3     # starts localstack for snowflake (snowflake image, 0.3 version)
localstack start -s localstack:4.4     # starts localstack for aws (localstack image, 4.4 version)
```

Heads-up: We are experimenting with the _Stacks_ terminology also in the Console, see https://github.com/localstack/localstack-web/pull/1787.

<!-- Optional section: How to test these changes? -->
<!--
## Testing

-->

<!-- Optional section: What's left to do before it can be merged? -->
<!--
## TODO

What's left to do:

- [ ] ...
- [ ] ...
-->

silv-io commented on localstack-core/localstack/cli/localstack.py:
```diff
@@ -472,6 +472,13 @@ def _print_service_table(services: Dict[str, str]) -> None:
+@click.option(
+    "--stack",
+    "-s",
+    type=str,
```
I'd use click.Choice here to enforce only `snowflake` can be used here

eruditmorina replied:
For passing specific versions next to the image, we can also introduce another cli option that would look like:

```
localstack start -s snowflake -v 0.3.0
```

This was also mentioned in #12529 briefly to get ideas around it. Wdyt?

gtsiolis replied:
Thanks @eruditmorina! I would still keep the {image}:{version} approach for simplicity, having a Docker-familiar syntax, not creating any confusion on what version (-v) is referring to (CLI, Image, etc). Let me know if I'm missing something!

| {image}:{version} | {image} -v {version} |
|--------|--------|
| <img width="686" alt="Frame 48097105" src="https://github.com/user-attachments/assets/df37041c-e060-4663-b369-16cce3921f43" /> | <img width="686" alt="Frame 48097106" src="https://github.com/user-attachments/assets/cb356ffe-743c-4e73-8c7b-dd70c895ea31" /> |

silv-io commented on localstack-core/localstack/cli/localstack.py:
```diff
@@ -472,6 +472,13 @@ def _print_service_table(services: Dict[str, str]) -> None:
+    "--stack",
+    "-s",
+    type=str,
+    help="Use a specific LocalStack stack with optional version.",
```
lgtm!

gtsiolis commented:
Rebased to run the pipeline with the latest changes from `master`. All checks passed. ✔️ 

Merged! :shipit: 

Thanks @silv-io and @eruditmorina for taking a look, inspiring this, and closing the draft PRs[[1](https://github.com/localstack/localstack/pull/12674)][[2](https://github.com/localstack/localstack/pull/12529)]! 

Added you both as co-authors. 🏂 🏂 🏂 

Thanks also everyone involved in the early discussions of adding a CLI command or option for running `snowflake`.
